ledger_device_sdk/src/lib.rs

```diff
@@ -52,6 +53,43 @@ macro_rules! set_panic {
     };
 }
 
+#[cfg(feature = "heap")]
+use critical_section::RawRestoreState;
+#[cfg(feature = "heap")]
+use embedded_alloc::Heap;
+
+#[cfg(feature = "heap")]
+#[global_allocator]
+static HEAP: Heap = Heap::empty();
+
+#[cfg(feature = "heap")]
+struct CriticalSection;
+#[cfg(feature = "heap")]
+critical_section::set_impl!(CriticalSection);
+
+/// Default empty implementation as we don't have concurrency.
+#[cfg(feature = "heap")]
+unsafe impl critical_section::Impl for CriticalSection {
+    unsafe fn acquire() -> RawRestoreState {}
+    unsafe fn release(restore_state: RawRestoreState) {}
+}
+
+/// Initializes the heap memory for the global allocator.
+///
+/// The heap is stored in the stack, and has a fixed size.
+/// This method is called just before [sample_main].
+#[no_mangle]
+#[cfg(feature = "heap")]
+extern "C" fn heap_init() {
+    const HEAP_SIZE: usize = 8192;
```

Comment:
Could we make this size configurable ?

Reply:
Indeed it would be better if this was configurable by the SDK users.

For the moment I am not sure how to make it configurable, I currently know two possibilities:
- pass the heap size as an environment variable during the compilation (defined in `.cargo/config.toml`), fetched in the SDK code using rust `env!` macro,
- use crate features and have predefined possible heap sizes (`heap-1k`, `heap-2k`, `heap-4k`, `heap-8k`, etc.).

First solution offers more flexibility, but seems to be a bit hacky. I find second option cleaner, but it restricts the possible heap sizes.

For the moment I suggest keeping a fixed value that should work for 90% of the cases, and see later how to improve that. If heap size is too big, users can still disable the feature and work without heap as of today.